copy.go

```diff
@@ -155,14 +149,23 @@ func fcopy(src, dest string, info os.FileInfo, opt Options) (err error) {
 	return
 }
 
+// fopen opens the named file,
+// and returns it regardless of its type
+// fs.File or *os.File
+func fopen(src string, opt Options) (io.ReadCloser, error) {
+	if opt.FS != nil {
+		return opt.FS.Open(src)
+	}
+	return os.Open(src)
+}
+
 // dcopy is for a directory,
 // with scanning contents inside the directory
 // and pass everything to "copy" recursively.
 func dcopy(srcdir, destdir string, info os.FileInfo, opt Options) (err error) {
-	if skip, err := onDirExists(opt, srcdir, destdir); err != nil {
-		return err
-	} else if skip {
-		return nil
+	skip, err := onDirExists(opt, srcdir, destdir)
+	if err != nil || skip {
+		return
 	}
```

Comment:
[Just IMO for readability] 🤔 
Returning error on `error` and skip this `dcopy` if on `skip == true` are different things. Thus I prefer separating those 2 in two returns.
What do you think?